### ISSUE TITLE
Got exception when using openOPC in python3.6,win10 x64

### ISSUE THREAD

**Issue by heroxxCN** [Alexhll/OpenOPC-python3.6#1]
>>> opc = OpenOPC.client()
Traceback (most recent call last):
  File "C:\Python36\lib\site-packages\win32com\client\dynamic.py", line 89, in _GetGoodDispatch
    IDispatch = pythoncom.connect(IDispatch)
pywintypes.com_error: (-2147221021, '操作无法使用', None, None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python36\lib\site-packages\OpenOPC.py", line 167, in __init__
    self._opc = win32com.client.gencache.EnsureDispatch(c, 0)
  File "C:\Python36\lib\site-packages\win32com\client\gencache.py", line 527, in EnsureDispatch
    disp = win32com.client.Dispatch(prog_id)
  File "C:\Python36\lib\site-packages\win32com\client\__init__.py", line 95, in Dispatch
    dispatch, userName = dynamic._GetGoodDispatchAndUserName(dispatch,userName,clsctx)
  File "C:\Python36\lib\site-packages\win32com\client\dynamic.py", line 114, in _GetGoodDispatchAndUserName
    return (_GetGoodDispatch(IDispatch, clsctx), userName)
  File "C:\Python36\lib\site-packages\win32com\client\dynamic.py", line 91, in _GetGoodDispatch
    IDispatch = pythoncom.CoCreateInstance(IDispatch, None, clsctx, pythoncom.IID_IDispatch)
pywintypes.com_error: (-2147221164, '没有注册类', None, None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\OpenOPC.py", line 173, in __init__
    raise OPCError(error_msg)
OpenOPC.OPCError: Dispatch: 没有注册类

**Comment by IMartin666** [comment]
I think you dont achieve the step 5 in [https://github.com/Alexhll/OpenOPC-python3.6](url)  ### C:\OpenOPC36\lib\regsvr32 gbda_aut.dll###   try it again and open the cmd with **Administrator status**

**Comment by svenpauline** [comment]
One additional note to add: When doing step 9, it  happened on my system (Win10, Python 3.6.6) with the following error:
        The OpenOPC Gateway Service service failed to start due to the following error: 
        The service did not respond to the start or control request in a timely fashion.
I found it is caused by missing a file named 'pywintypes36.dll' under 'C:\Users\PC\AppData\Local\Programs\Python\Python36\Lib\site-packages\win32'. This file can be found in 'C:\Users\PC\AppData\Local\Programs\Python\Python36\Lib\site-packages\pywin32_system32'. After copying it to the above folder the service can start.
I think it helpful to add that information as well.

**Comment by svenpauline** [comment]
Unfortunately, i can't get it working after making step 9. The service can start but when executing the python file i have the error: (win10, python 3.6.6, working in open mode (not dcom))

o.connect('localhost')
Traceback (most recent call last):

  File "<ipython-input-4-f82f99b94649>", line 1, in <module>
    o.connect('localhost')

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\core.py", line 186, in __call__
    return self.__send(self.__name, args, kwargs)

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\core.py", line 461, in _pyroInvoke
    data = serializer.deserializeData(msg.data, compressed=msg.flags & message.FLAGS_COMPRESSED)

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\util.py", line 170, in deserializeData
    return self.loads(data)

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\util.py", line 558, in loads
    return self.recreate_classes(serpent.loads(data))

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\util.py", line 411, in recreate_classes
    return self.dict_to_class(literal)

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\util.py", line 575, in dict_to_class
    return super(SerpentSerializer, cls).dict_to_class(data)

  File "C:\Users\PC\Anaconda3\lib\site-packages\Pyro4\util.py", line 390, in dict_to_class
    raise errors.SerializeError("unsupported serialized class: " + classname)

SerializeError: unsupported serialized class: OpenOPC.OPCError

I made it work with Python 2.7 (in open mode) but I can't make it work in Python 3.6. The OPC server, OPC server configuration and environment variables are exactly the same (all in open mode). Even the python code are the same. Any clues?

**Comment by ydwpython** [comment]
> 我认为你没有实现[https://github.com/Alexhll/OpenOPC-python3.6](url) ### C：\ OpenOPC36 \ lib \ regsvr32 gbda_aut.dll ###中的第5步再次尝试并使用**Administrator**打开cmd **状态**

你好，，你是北京的吗，能联系到你吗

**Comment by heroxxCN** [comment]
> I think you dont achieve the step 5 in [https://github.com/Alexhll/OpenOPC-python3.6](url) ### C:\OpenOPC36\lib\regsvr32 gbda_aut.dll### try it again and open the cmd with **Administrator status**

I've did it exactly as step 5,but it did not work on win10x64。
I've got it worked well on win7 x32.

**Comment by 2makeitwork** [comment]
Please be aware the gbda_aut.ll has a 32bit and 64bit version. I can happen you didn't register the right 'bit' version.
However, I never made it working under Win10 64bit but for a different reason. 
I use the 'open' mode and can use the .server(), .list() from a remote Linux without any issue. But once I use the .read('variable') function, the zzzOpenOPCService automatically 'stops' and I receive a connectionClosedError from Pyro4.

However, I use the same configuration on a Windows 7 74bit in open mode without any issue. I also found the client and server side must be the same 'bit'ness, although I only tested on 64bit.

To sum up, what works is:
OPC Server side: Win7 64bit, Python 3.7.2 64bit
Client side: Ubuntu Stretch 64bit, Python 3.5.3 64bit

**Comment by 2makeitwork** [comment]
update:
After I restarted win10, the openOPCGatewayService cannot be started, after you click start, it immediately stopped with a pop up message: the OpenOPC Gateway Service service on Local Computer started and then stopped. Some services stop automatically if they are not in use by other services or programs.
